src/data/markdown/translated-guides/en/02 Using k6/00 Tutorial/400 Reuse and re-run tests.md

```diff
@@ -0,0 +1,410 @@
+const workload = JSON.parse(open("./stress.json"));
+console.log(workload.stress_test);
+export const options = {
+  //use converted JSON config
+  scenarios: workload,
+  //use imported thresholds variable
+  thresholds,
+};
+
+// use globalThis to set global variables
+globalThis.baseUrl = "https://test.k6.io";
+
+// Function to test user flow
+export default function () {
+  // Put visits to contact page in one group
+  contacts();
+  // Coinflip players in another group
+  coinflip();
+}
+```
+
+```javascript
+export const thresholds = {
+  thresholds: {
+    http_req_failed: [{ threshold: "rate<0.01", abortOnFail: true }], // availability threshold for error rate
+    http_req_duration: ["p(99)<1000"], // Latency threshold for percentile
+  },
+};
+```
+
+```json
+{
+  "stress_test": {
+    "executor": "ramping-arrival-rate",
+    "startRate": 20,
+    "timeUnit": "1m",
+    "preAllocatedVUs": 200,
+    "stages": [
+      { "target": 20, "duration": "10s" },
+      { "target": 40, "duration": "10s" },
+      { "target": 40, "duration": "10s" },
+      { "target": 60, "duration": "10s" },
+      { "target": 60, "duration": "30s" },
+      { "target": 0, "duration": "1m" }
+    ]
+  }
+}
+```
+
+</CodeGroup>
+
+Notice the length of this final script and compare it with the script at the beginning of this page.
+Though the final execution is the same, it's half the size and more readable.
+
+Besides shortness, this modularity lets you compose scripts from many parts, or dynamically configure scripts at run time.
+
+
+
+## Mix and match logic
+
+With modularized configuration and logic, you can mix and match logic.
+An easy way to configure this is through [environment variables](/using-k6/environment-variables).
+
+Change `main.js` so that it:
+- _By default_ runs a smoke test with 5 iterations
+- With the right enviroment variables, runs a stress test.
+
+To do this, follow these steps:
+
+1. Add a new scenario file, `smoke.json`, with the following configuration
+
+  <CodeGroup labels={["smoke.json"]} lineNumbers={[]} showCopyButton={[true]}>
+
+  ```
+  {
+    "smoke": {
+      "executor": "shared-iterations",
+      "iterations": 5,
+      "vus": 1,
+      "exec": "scenario1"
+    }
+  }
+  ```
+
+  </CodeGroup>
+
+1. Add conditional logic to bind `workload` to `__ENV.WORKLOAD`  if the environment variable exists, and to the content of `smoke.json` if it doesn't.
+  
+  ```javascript
+  let workload;
+  if (typeof __ENV.WORKLOAD !== "undefined") {
+    workload = JSON.parse(open(__ENV.WORKLOAD));
+  } else {
+    //if no envvar, use smoke config
+    workload = JSON.parse(open("./smoke.json"));
+  }
+  ```
+  
+1. Run the script with and without the `-e` flag.
+
+   - What happens when you run `k6 run main.js`?
+   - What about `k6 run main.js -e WORKLOAD="./stress.json"`?
+
+If its an object, you can modularize it. Since k6 scripts are JavaScript, where essentially everything is an object, you can modularize essentially everything. 
```

Comment:
```suggestion
If it's an object, you can modularize it. Since k6 scripts are JavaScript, where everything is an object, you can modularize everything. 
```